corehq/apps/locations/static/locations/js/location_tree.async.js

```diff
@@ -1,410 +1,455 @@
-/* globals django, reloadLocationSearchSelect */
-// reloadLocationSearchSelect is defined in locations/manage/locations.html
-
-function api_get_children(loc_uuid, callback) {
-    var params = (loc_uuid ? {
-        parent_id: loc_uuid,
-    } : {});
-    // show_inactive comes from global state
-    params.include_inactive = show_inactive;
-    $.getJSON(LOAD_LOCS_URL, params, function(allData) {
-        callback(allData.objects);
-    });
-}
-
-function LocationTreeViewModel(hierarchy) {
-    var model = this;
-
-    this.root = ko.observable();
-
-    // TODO this should reference location type settings for domain
-    this.location_types = $.map(hierarchy, function(e) {
-        return {
-            type: e[0],
-            allowed_parents: e[1],
-        };
-    });
-
-    // search for a location within the tree by uuid; return path to location if found
-    this.find_loc = function(uuid, loc) {
-        loc = loc || this.root();
-
-        if (loc.uuid() === uuid) {
-            return [loc];
-        } else {
-            var path = null;
-            $.each(loc.children(), function(i, e) {
-                var subpath = model.find_loc(uuid, e);
-                if (subpath) {
-                    path = subpath;
-                    path.splice(0, 0, loc);
-                    return false;
-                }
-            });
-            return path;
-        }
-    };
+/* globals django */
+hqDefine('locations/js/location_tree', function() {
+    function api_get_children(loc_uuid, show_inactive, load_locs_url, callback) {
+        var params = (loc_uuid ? {
+            parent_id: loc_uuid,
+        } : {});
+        // show_inactive comes from global state
+        params.include_inactive = show_inactive;
+        $.getJSON(load_locs_url, params, function(allData) {
+            callback(allData.objects);
+        });
+    }
+
+    function LocationTreeViewModel(hierarchy, options) {
+        // options should have properties:
```

Comment:
This is a nice addition - we have a thousand places where an `options` object is passed in and rarely do any validation on what's in that object, it'd be good to get a pattern going eventually.

Reply:
Yeah, I would much prefer to switch these over to an 'options/config object' at some point with a set of inputs, but this seemed to work for now